middleware/etcd/lookup_test.go

```diff
@@ -0,0 +1,397 @@
+// +build net
+
+package etcd
+
+// etcd needs to be running on http://127.0.0.1:2379
+// *and* needs connectivity to the internet for remotely resolving
+// names.
+
+import (
+	"encoding/json"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/miekg/coredns/middleware"
+	"github.com/miekg/coredns/middleware/etcd/msg"
+	"github.com/miekg/coredns/middleware/etcd/singleflight"
+	"github.com/miekg/coredns/middleware/proxy"
+	"github.com/miekg/dns"
+
+	etcdc "github.com/coreos/etcd/client"
+	"golang.org/x/net/context"
+)
+
+var (
+	etc    Etcd
+	client etcdc.KeysAPI
+	ctx    context.Context
+)
+
+type Section int
+
+const (
+	Answer Section = iota
+	Ns
+	Extra
+)
+
+func init() {
+	ctx = context.TODO()
+
+	etcdCfg := etcdc.Config{
+		Endpoints: []string{"http://localhost:2379"},
```

Comment:
[ Quoting <notifications@github.com> in "Re: [coredns] Etcd (#34)..." ]
>#shameless plug:
>https://github.com/mwitkow/go-etcd-harness

Ah nice. I've looked around for *shrudder* a etcd mock, but this is the next
best thing (and there are weird-ish things in etcd's impl. that a mock prolly
won't catch).

Def. worth using here